### PR TITLE
[7X BKPT] Tolerate failure of bad buffer flushing on a temporary table.

### PR DESCRIPTION
**Pull request by haolinw**
This is backporting https://github.com/greenplum-db/gpdb/pull/14949.

This is intended to fix the issue #13809 (see also #6351). The problem is, when the writer aborts the transaction before any of readers, it drops the shared_buffer and unlinks corresponding temporary relfilenodes. But readers may be yet to receive the cancel signal from QD hence continue executing their part of the plan. This makes a reader has a chance to read previous unlinked relfilenode into shared_buffers and re-mark it to dirty in the case of hintbit is set, which could result to a permanent failure could not open file when other processes (such like bgworker or readers) attempt to evict or flush this dirty buffer to disk.

The patch is trying to address a kinda like semantic contradiction that a temporary object should not generate a permanent failure to block later queries permanently. Given above context, the patch clears the dirty shared buffer of the non-existent temporary relfilenode to allow queries to be proceeded as is. Those queries could be finally failed because of accessing that non-existent file but that just generates a transient failure rather than a permanent failure, hence DB instance could be continue working.

## Here are some reminders before you submit the pull request
- [ ] Add tests for the change
- [ ] Document changes
- [ ] Communicate in the mailing list if needed
- [ ] Pass `make installcheck`
- [ ] Review a PR in return to support the community

### CODE REVIEW

File: src/backend/access/appendonly/appendonlyam.c [hunk 1]
```diff
@@ -2833,9 +2833,12 @@ appendonly_insert(AppendOnlyInsertDesc aoInsertDesc,
 	 */
 	if (aoInsertDesc->numSequences == 0)
 	{
-		int64 firstSequence = GetFastSequences(aoInsertDesc->segrelid,
```
Comment (ashwinstar) [inline src/backend/access/appendonly/appendonlyam.c:2836]: This commit shouldn't be part of this PR. Plus, PostgreSQL and greenplum have special macro(not on computer to have handy name) to attribute for variables which are assert only which we should use here I think instead.

Comment (haolinw) [reply src/backend/access/appendonly/appendonlyam.c:2836]: OK, will move this change to another PR and use PG_USED_FOR_ASSERTS_ONLY instead.

**Review by interma — APPROVED** [review]
LGTM
